src/SCRIPTS/TELEMETRY/iNav/elrs.lua

```diff
@@ -3,54 +3,62 @@
 local deviceId = 0xEE
 
 local function byteToStr(b)
-	-- Translate b into a string from symbolChars if available, else use string.char
-	return symbolChars and symbolChars[b] or string.char(b)
+   -- Translate b into a string from symbolChars if available, else use string.char
+   return symbolChars and symbolChars[b] or string.char(b)
 end
 
 local function fieldGetString(data, offset, last)
-	if last then
-		return fieldStrFF(data, offset, last)
-	end
+   if last then
+      return fieldStrFF(data, offset, last)
+   end
 
-	local result = ""
-	while data[offset] ~= 0 do
-		result = result .. byteToStr(data[offset])
-		offset = offset + 1
-	end
+   local result = ""
+   while data[offset] ~= 0 do
+      result = result .. byteToStr(data[offset])
+      offset = offset + 1
+   end
 
-	return result, offset + 1
+   return result, offset + 1
 end
 
 local function fieldGetValue(data, offset, size)
-	local result = 0
-	for i=0, size-1 do
-		result = bit32.lshift(result, 8) + data[offset + i]
-	end
-	return result
+   local result = 0
+   for i=0, size-1 do
+      result = bit32.lshift(result, 8) + data[offset + i]
+   end
+   return result
 end
 
 local function parseDeviceInfoMessage(data)
-	local offset
-	local id = data[2]
-	local devicesName
-	devicesName, offset = fieldGetString(data, 3)
-	if deviceId == id then
-		deviceIsELRS_TX = ((fieldGetValue(data,offset,4) == 0x454C5253) and (deviceId == 0xEE)) or nil -- SerialNumber = 'E L R S' and ID is TX module
-		return deviceIsELRS_TX
-	end
+   local offset
+   local id = data[2]
+   local devicesName
+   local majId = -1
+   devicesName, offset = fieldGetString(data, 3)
+   if deviceId == id then
+      if ((fieldGetValue(data,offset,4) == 0x454C5253) and (deviceId == 0xEE)) then
+	 majId = data[offset+8] - 0x30
```

Comment:
This is not the correct data interpretation.  Look how the version is set in Expresslrs CRSF::GetDeviceInformation.

Captures I have prior to elrsv3 don't have the version set (all zeroes), so maybe just check for version 3.  But I can't find the code that sends it in elrs at the moment and out of time to look :)

Here's a device info packet from 3.0.1: `ea 1c 29 ea ee 48 4d 20 45 53 32 34 54 58 00 45 4c 52 53 00 00 00 00 00 03 00 01 1b 00 c9 [Device] 0xee "HM ES24TX^@" 27 parameters`